hare/eligibility/oracle.go

```diff
@@ -231,12 +261,17 @@ func (o *Oracle) actives(layer types.LayerID) (map[string]struct{}, error) {
 		return nil, errGenesis
 	}
 
+	// check cache
+	if val, exist := o.activesCache.Get(safeEp); exist {
+		return val.(map[string]struct{}), nil
+	}
+
 	// lock until any return
 	// note: no need to lock per safeEp - we do not expect many concurrent requests per safeEp (max two)
 	o.lock.Lock()
 
 	// check cache
-	if val, exist := o.cache.Get(safeEp); exist {
+	if val, exist := o.activesCache.Get(safeEp); exist {
```

Comment:
haven't you jut checked this above outside the lock ?

Reply:
Yes, it is an optimization, though after rethinking it, it is probably negligible.

Reply:
I removed it from both the activesCache and the vrfMsgCache